File: Java/58_LengthLastWord/Solution.java
public class Solution {
  public int lengthOfLastWord(String s) {
    int length=0;
    for(int i = 0; i < s.length();)
      if(s.charAt(i++) != ' ') length++;
      else if (i < s.length() && s.charAt(i) != ' ') length = 0;
    return length;
  }

  public static void main (String [] args) {
    String s = "   fly me   to   the moon  ";
    Solution solution = new Solution();
    System.out.println(solution.lengthOfLastWord(s));
  }
}

File: C++/58_LengthLastWord/Solution.cpp
#include<iostream>
#include<cstring>

int lengthOfLastWord(std::string s) {
  int length = 0;
  for(int i = 0; s[i];)
    if(s[i++] != ' ') length++;
    else if (s[i] && s[i] != ' ') length=0;
  return length;
}

int main (void) {
  std::string s = "   fly me   to   the moon  ";
  std::cout << lengthOfLastWord(s) << std::endl;
  return 0;
}